tests/unit/core/git_hooks/ci/test_previous_commit.py

```diff
@@ -121,3 +127,93 @@ def test_get_previous_commit_from_ci_env_new_repo(
         # Simulate CI env
         setup_ci_env(monkeypatch, "new-branch", "HEAD")
         assert get_previous_commit_from_ci_env(False) is None
+
+
+@parametrized_ci_provider
+def test_get_previous_commit_from_ci_env_sub_branch_forced_push(
+    monkeypatch: MonkeyPatch,
+    tmp_path: Path,
+    setup_ci_env: Callable[[MonkeyPatch, str, str], None],
+) -> None:
+    """
+    GIVEN   a repository with:
+            - a commit pushed on the main branch
+            - another branch created from main, with commits pushed
+            - a local history rewritten
+    WHEN    calculating the commit sha previous to the forced push,
+            in a simulated CI env where new-branch is newly pushed
+    THEN    the commit SHA (previous HEAD) returned does not exist anymore
+            and a scan all is triggered instead
+    """
+    # Setup main branch
+    remote_repository = Repository.create(tmp_path / "remote", bare=True)
+    repository = Repository.clone(remote_repository.path, tmp_path / "local")
+    repository.create_commit()
+    repository.push()
+
+    # Setup new branch
+    branch_name = "new-branch"
+    repository.create_branch(branch_name)
+    before_sha = repository.create_commit()
+
+    repository.git("commit", "--amend", "--allow-empty", "-m", "message")
+    repository.remove_unreachable_commits()
+
+    with cd(repository.path), mock.patch.dict(os.environ, clear=True):
+        # Simulate CI env
+        setup_ci_env(monkeypatch, branch_name, "HEAD", before_sha)
+        found_sha = get_previous_commit_from_ci_env(False)
+
+    assert found_sha is None
+
+
+@parametrized_ci_provider
+def test_get_previous_commit_from_ci_env_default_branch_forced_push(
+    monkeypatch: MonkeyPatch,
+    tmp_path: Path,
+    setup_ci_env: Callable[[MonkeyPatch, str, str], None],
+):
+    """
+    GIVEN   a new, empty repository
+    WHEN    calculating the commit sha previous to the first push,
+            in a simulated CI env
+    THEN    None is returned
+    """
+    remote_repository = Repository.create(tmp_path / "remote", bare=True)
+    repository = Repository.clone(remote_repository.path, tmp_path / "local")
+
+    repository.create_commit()
+    before_sha = repository.create_commit()
+
+    repository.git("commit", "--amend", "--allow-empty", "-m", "message")
+    repository.remove_unreachable_commits()
+
```

Comment:
nit: assert here (and in the other force push tests) that:
- `is_valid_git_commit_ref(before_sha)` is False (= we properly set an invalid SHA in CI env, which is what Github/Gitlab do)
- `repository.get_top_sha()` is different from `before_sha` (= the repository is ready for a force push)